### ISSUE TITLE
bug:  Controller Manager fails to install on OpenShift - Times out waiting for CRD install

### ISSUE THREAD

**Issue by cwiklik** [kubestellar/kubestellar#1916]
### Describe the bug

I tried to install Kubestellar (plus Kubeflex) on OpenShift using 0.21.0 release. The Controller Manager install fails with 

I0319 14:59:40.253321       1 crds.go:76] "applying crd" logger="Binding" name="bindings.control.kubestellar.io"
E0319 14:59:40.268791       1 main.go:147] "error installing the CRDs" err="context deadline exceeded" logger="setup"

This install uses WDS of type HOST

I use this command to install the Controller Manager:

helm  upgrade --install -n wds2-system kubestellar oci://ghcr.io/kubestellar/kubestellar/controller-manager-chart --version 0.21.0 --set ControlPlaneName=wds2 --set APIGroups=workload.codeflare.dev

The full log is here:
[cm.log](https://github.com/kubestellar/kubestellar/files/14652779/cm.log)

### Steps To Reproduce

kflex init

kubectl apply -f https://raw.githubusercontent.com/kubestellar/kubestellar/v0.21.0/config/postcreate-hooks/kubestellar.yaml
kubectl apply -f https://raw.githubusercontent.com/kubestellar/kubestellar/v0.21.0/config/postcreate-hooks/ocm.yaml

kflex create imbs1 --type vcluster -p ocm

helm --kube-context imbs1 upgrade --install status-addon -n open-cluster-management oci://ghcr.io/kubestellar/ocm-status-addon-chart --version v0.2.0-rc7

kflex create wds1 -p kubestellar

clusteradm --context imbs1 get token

On the WEC I run:
clusteradm join --hub-token <TOKEN> --hub-apiserver https://<hub endpoint> --cluster-name wec3

On the HUB I run:
clusteradm --context imbs1 accept --clusters wec3

kubectl config rename-context default/c109-e-eu-gb-containers-cloud-ibm-com:31392/system:serviceaccount:default:my-service-account wec3

// Create ClusterBinding
kubectl --context default/c115-e-us-south-containers-cloud-ibm-com:30829/system:serviceaccount:default:my-service-account apply -f - <<
EOF
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: kubeflex-manager-cluster-admin-rolebinding
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
- kind: ServiceAccount
  name: kubeflex-controller-manager
  namespace: kubeflex-system
EOF

kubectl label controlplane wds2 kflex.kubestellar.io/cptype=wds

kflex ctx

helm  upgrade --install -n wds2-system kubestellar oci://ghcr.io/kubestellar/kubestellar/controller-manager-chart --version 0.21.0 --set ControlPlaneName=wds2 --set APIGroups=workload.codeflare.dev

kubectl get deployments.apps -n wds2-system kubestellar-controller-manager

### Expected Behavior

Controller Manager installs and in Ready state.

**Comment by pdettori** [comment]
This seems to be a re-occurence of #1705

**Comment by waltforme** [comment]
I quickly looked at the log, sounds to me like a different issue from #1705.

#1705 's root cause is the informers' cache syncing and initial reconciliation on `CustomResourceDefinition` objects.
This one's throttling, on the other hand, seems to be _before_ the creation of informers.

A few things I want to look at:
1. The discovery during the creation of KS CRDs. I think this one can be refactored/simplified:
    https://github.com/kubestellar/kubestellar/blob/7a2901660073483a206e4b82683e82a9bcb2f276/pkg/crd/crds.go#L94
2. The version of the OCP. I doubt it is using legacy discovery.

To summarize I think this one and #1705 appear to be both throttling but, but with different root cause. The title of this issue ('waiting for CRD install') is quite accurate. I will further investigate this issue.

**Comment by pdettori** [comment]
@waltforme I agree, I did some preliminary investigation as well and the issue happens during discovery in the CRD code as you pointed out. One quick solution that I tried was just to increase context duration. in https://github.com/kubestellar/kubestellar/blob/755304114a53a09cc26c23148e7e39ba0afad8c6/pkg/crd/crds.go#L60, however it would be best to optimize the discovery part. I also managed (with some minor change to the main) to run the code as bare process from my machine to the same cluster. For some reason I could not explain the behavior is different than when running as a pod, and there is much less throttling. Do you know if client-side throttling is affected by running in-cluster vs. off-cluster?

**Comment by waltforme** [comment]
@pdettori I see, increasing the context timeout is a good point 👍. I think we can/should do both, optimize the discovery and make the timeout longer.

I could not explain why there are differences between running in-cluster v.s. off-cluster, either 🤔.
